File: src/java/src/quickfix/fix40/Message.java
package quickfix.fix40;
import quickfix.FieldNotFound;
import quickfix.Group;
import quickfix.field.*;

public class Message extends quickfix.Message {
  public Message() {
    super();
    header = new Header();
    trailer = new Trailer();
    getHeader().setField(new BeginString("FIX.4.0"));
  }
  public class Header extends quickfix.Message.Header {
      public void set(quickfix.field.BeginString value)
      { setField(value); }
      public quickfix.field.BeginString get(quickfix.field.BeginString  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.BeginString getBeginString() throws FieldNotFound
      { quickfix.field.BeginString value = new quickfix.field.BeginString();
        getField(value); return value; }
      public boolean isSet(quickfix.field.BeginString field)
      { return isSetField(field); }
      public boolean isSetBeginString()
      { return isSetField(8); }
      public void set(quickfix.field.BodyLength value)
      { setField(value); }
      public quickfix.field.BodyLength get(quickfix.field.BodyLength  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.BodyLength getBodyLength() throws FieldNotFound
      { quickfix.field.BodyLength value = new quickfix.field.BodyLength();
        getField(value); return value; }
      public boolean isSet(quickfix.field.BodyLength field)
      { return isSetField(field); }
      public boolean isSetBodyLength()
      { return isSetField(9); }
      public void set(quickfix.field.MsgType value)
      { setField(value); }
      public quickfix.field.MsgType get(quickfix.field.MsgType  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.MsgType getMsgType() throws FieldNotFound
      { quickfix.field.MsgType value = new quickfix.field.MsgType();
        getField(value); return value; }
      public boolean isSet(quickfix.field.MsgType field)
      { return isSetField(field); }
      public boolean isSetMsgType()
      { return isSetField(35); }
      public void set(quickfix.field.SenderCompID value)
      { setField(value); }
      public quickfix.field.SenderCompID get(quickfix.field.SenderCompID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.SenderCompID getSenderCompID() throws FieldNotFound
      { quickfix.field.SenderCompID value = new quickfix.field.SenderCompID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.SenderCompID field)
      { return isSetField(field); }
      public boolean isSetSenderCompID()
      { return isSetField(49); }
      public void set(quickfix.field.TargetCompID value)
      { setField(value); }
      public quickfix.field.TargetCompID get(quickfix.field.TargetCompID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.TargetCompID getTargetCompID() throws FieldNotFound
      { quickfix.field.TargetCompID value = new quickfix.field.TargetCompID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.TargetCompID field)
      { return isSetField(field); }
      public boolean isSetTargetCompID()
      { return isSetField(56); }
      public void set(quickfix.field.OnBehalfOfCompID value)
      { setField(value); }
      public quickfix.field.OnBehalfOfCompID get(quickfix.field.OnBehalfOfCompID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.OnBehalfOfCompID getOnBehalfOfCompID() throws FieldNotFound
      { quickfix.field.OnBehalfOfCompID value = new quickfix.field.OnBehalfOfCompID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.OnBehalfOfCompID field)
      { return isSetField(field); }
      public boolean isSetOnBehalfOfCompID()
      { return isSetField(115); }
      public void set(quickfix.field.DeliverToCompID value)
      { setField(value); }
      public quickfix.field.DeliverToCompID get(quickfix.field.DeliverToCompID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.DeliverToCompID getDeliverToCompID() throws FieldNotFound
      { quickfix.field.DeliverToCompID value = new quickfix.field.DeliverToCompID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.DeliverToCompID field)
      { return isSetField(field); }
      public boolean isSetDeliverToCompID()
      { return isSetField(128); }
      public void set(quickfix.field.SecureDataLen value)
      { setField(value); }
      public quickfix.field.SecureDataLen get(quickfix.field.SecureDataLen  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.SecureDataLen getSecureDataLen() throws FieldNotFound
      { quickfix.field.SecureDataLen value = new quickfix.field.SecureDataLen();
        getField(value); return value; }
      public boolean isSet(quickfix.field.SecureDataLen field)
      { return isSetField(field); }
      public boolean isSetSecureDataLen()
      { return isSetField(90); }
      public void set(quickfix.field.SecureData value)
      { setField(value); }
      public quickfix.field.SecureData get(quickfix.field.SecureData  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.SecureData getSecureData() throws FieldNotFound
      { quickfix.field.SecureData value = new quickfix.field.SecureData();
        getField(value); return value; }
      public boolean isSet(quickfix.field.SecureData field)
      { return isSetField(field); }
      public boolean isSetSecureData()
      { return isSetField(91); }
      public void set(quickfix.field.MsgSeqNum value)
      { setField(value); }
      public quickfix.field.MsgSeqNum get(quickfix.field.MsgSeqNum  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.MsgSeqNum getMsgSeqNum() throws FieldNotFound
      { quickfix.field.MsgSeqNum value = new quickfix.field.MsgSeqNum();
        getField(value); return value; }
      public boolean isSet(quickfix.field.MsgSeqNum field)
      { return isSetField(field); }
      public boolean isSetMsgSeqNum()
      { return isSetField(34); }
      public void set(quickfix.field.SenderSubID value)
      { setField(value); }
      public quickfix.field.SenderSubID get(quickfix.field.SenderSubID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.SenderSubID getSenderSubID() throws FieldNotFound
      { quickfix.field.SenderSubID value = new quickfix.field.SenderSubID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.SenderSubID field)
      { return isSetField(field); }
      public boolean isSetSenderSubID()
      { return isSetField(50); }
      public void set(quickfix.field.TargetSubID value)
      { setField(value); }
      public quickfix.field.TargetSubID get(quickfix.field.TargetSubID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.TargetSubID getTargetSubID() throws FieldNotFound
      { quickfix.field.TargetSubID value = new quickfix.field.TargetSubID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.TargetSubID field)
      { return isSetField(field); }
      public boolean isSetTargetSubID()
      { return isSetField(57); }
      public void set(quickfix.field.OnBehalfOfSubID value)
      { setField(value); }
      public quickfix.field.OnBehalfOfSubID get(quickfix.field.OnBehalfOfSubID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.OnBehalfOfSubID getOnBehalfOfSubID() throws FieldNotFound
      { quickfix.field.OnBehalfOfSubID value = new quickfix.field.OnBehalfOfSubID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.OnBehalfOfSubID field)
      { return isSetField(field); }
      public boolean isSetOnBehalfOfSubID()
      { return isSetField(116); }
      public void set(quickfix.field.DeliverToSubID value)
      { setField(value); }
      public quickfix.field.DeliverToSubID get(quickfix.field.DeliverToSubID  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.DeliverToSubID getDeliverToSubID() throws FieldNotFound
      { quickfix.field.DeliverToSubID value = new quickfix.field.DeliverToSubID();
        getField(value); return value; }
      public boolean isSet(quickfix.field.DeliverToSubID field)
      { return isSetField(field); }
      public boolean isSetDeliverToSubID()
      { return isSetField(129); }
      public void set(quickfix.field.PossDupFlag value)
      { setField(value); }
      public quickfix.field.PossDupFlag get(quickfix.field.PossDupFlag  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.PossDupFlag getPossDupFlag() throws FieldNotFound
      { quickfix.field.PossDupFlag value = new quickfix.field.PossDupFlag();
        getField(value); return value; }
      public boolean isSet(quickfix.field.PossDupFlag field)
      { return isSetField(field); }
      public boolean isSetPossDupFlag()
      { return isSetField(43); }
      public void set(quickfix.field.PossResend value)
      { setField(value); }
      public quickfix.field.PossResend get(quickfix.field.PossResend  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.PossResend getPossResend() throws FieldNotFound
      { quickfix.field.PossResend value = new quickfix.field.PossResend();
        getField(value); return value; }
      public boolean isSet(quickfix.field.PossResend field)
      { return isSetField(field); }
      public boolean isSetPossResend()
      { return isSetField(97); }
      public void set(quickfix.field.SendingTime value)
      { setField(value); }
      public quickfix.field.SendingTime get(quickfix.field.SendingTime  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.SendingTime getSendingTime() throws FieldNotFound
      { quickfix.field.SendingTime value = new quickfix.field.SendingTime();
        getField(value); return value; }
      public boolean isSet(quickfix.field.SendingTime field)
      { return isSetField(field); }
      public boolean isSetSendingTime()
      { return isSetField(52); }
      public void set(quickfix.field.OrigSendingTime value)
      { setField(value); }
      public quickfix.field.OrigSendingTime get(quickfix.field.OrigSendingTime  value)
        throws FieldNotFound
      { getField(value); return value; }
      public quickfix.field.OrigSendingTime getOrigSendingTime() throws FieldNotFound
      { quickfix.field.OrigSendingTime value = new quickfix.field.OrigSendingTime();
        getField(value); return value; }
      public boolean isSet(quickfix.field.OrigSendingTime field)
      { return isSetField(field); }
      public boolean isSetOrigSendingTime()
      { return isSetField(122); }
  }

    public void set(quickfix.field.SignatureLength value)
    { setField(value); }
    public quickfix.field.SignatureLength get(quickfix.field.SignatureLength  value)
      throws FieldNotFound
    { getField(value); return value; }
    public quickfix.field.SignatureLength getSignatureLength() throws FieldNotFound
    { quickfix.field.SignatureLength value = new quickfix.field.SignatureLength();
      getField(value); return value; }
    public boolean isSet(quickfix.field.SignatureLength field)
    { return isSetField(field); }
    public boolean isSetSignatureLength()
    { return isSetField(93); }
    public void set(quickfix.field.Signature value)
    { setField(value); }
    public quickfix.field.Signature get(quickfix.field.Signature  value)
      throws FieldNotFound
    { getField(value); return value; }
    public quickfix.field.Signature getSignature() throws FieldNotFound
    { quickfix.field.Signature value = new quickfix.field.Signature();
      getField(value); return value; }
    public boolean isSet(quickfix.field.Signature field)
    { return isSetField(field); }
    public boolean isSetSignature()
    { return isSetField(89); }
    public void set(quickfix.field.CheckSum value)
    { setField(value); }
    public quickfix.field.CheckSum get(quickfix.field.CheckSum  value)
      throws FieldNotFound
    { getField(value); return value; }
    public quickfix.field.CheckSum getCheckSum() throws FieldNotFound
    { quickfix.field.CheckSum value = new quickfix.field.CheckSum();
      getField(value); return value; }
    public boolean isSet(quickfix.field.CheckSum field)
    { return isSetField(field); }
    public boolean isSetCheckSum()
    { return isSetField(10); }
}

File: src/.NET/fix44/Message.cs
namespace QuickFix44
{
  public class Header : QuickFix.Message.Header
  {
    Header(QuickFix.Message message) : base(message) {}

      public void set(QuickFix.BeginString value)
      { setField(value); }
      public QuickFix.BeginString get(QuickFix.BeginString  value)
      { getField(value); return value; }
      public QuickFix.BeginString getBeginString()
      { QuickFix.BeginString value = new QuickFix.BeginString();
        getField(value); return value; }
      public bool isSet(QuickFix.BeginString field)
      { return isSetField(field); }
      public bool isSetBeginString()
      { return isSetField(8); }

      public void set(QuickFix.BodyLength value)
      { setField(value); }
      public QuickFix.BodyLength get(QuickFix.BodyLength  value)
      { getField(value); return value; }
      public QuickFix.BodyLength getBodyLength()
      { QuickFix.BodyLength value = new QuickFix.BodyLength();
        getField(value); return value; }
      public bool isSet(QuickFix.BodyLength field)
      { return isSetField(field); }
      public bool isSetBodyLength()
      { return isSetField(9); }

      public void set(QuickFix.MsgType value)
      { setField(value); }
      public QuickFix.MsgType get(QuickFix.MsgType  value)
      { getField(value); return value; }
      public QuickFix.MsgType getMsgType()
      { QuickFix.MsgType value = new QuickFix.MsgType();
        getField(value); return value; }
      public bool isSet(QuickFix.MsgType field)
      { return isSetField(field); }
      public bool isSetMsgType()
      { return isSetField(35); }

      public void set(QuickFix.SenderCompID value)
      { setField(value); }
      public QuickFix.SenderCompID get(QuickFix.SenderCompID  value)
      { getField(value); return value; }
      public QuickFix.SenderCompID getSenderCompID()
      { QuickFix.SenderCompID value = new QuickFix.SenderCompID();
        getField(value); return value; }
      public bool isSet(QuickFix.SenderCompID field)
      { return isSetField(field); }
      public bool isSetSenderCompID()
      { return isSetField(49); }

      public void set(QuickFix.TargetCompID value)
      { setField(value); }
      public QuickFix.TargetCompID get(QuickFix.TargetCompID  value)
      { getField(value); return value; }
      public QuickFix.TargetCompID getTargetCompID()
      { QuickFix.TargetCompID value = new QuickFix.TargetCompID();
        getField(value); return value; }
      public bool isSet(QuickFix.TargetCompID field)
      { return isSetField(field); }
      public bool isSetTargetCompID()
      { return isSetField(56); }

      public void set(QuickFix.OnBehalfOfCompID value)
      { setField(value); }
      public QuickFix.OnBehalfOfCompID get(QuickFix.OnBehalfOfCompID  value)
      { getField(value); return value; }
      public QuickFix.OnBehalfOfCompID getOnBehalfOfCompID()
      { QuickFix.OnBehalfOfCompID value = new QuickFix.OnBehalfOfCompID();
        getField(value); return value; }
      public bool isSet(QuickFix.OnBehalfOfCompID field)
      { return isSetField(field); }
      public bool isSetOnBehalfOfCompID()
      { return isSetField(115); }

      public void set(QuickFix.DeliverToCompID value)
      { setField(value); }
      public QuickFix.DeliverToCompID get(QuickFix.DeliverToCompID  value)
      { getField(value); return value; }
      public QuickFix.DeliverToCompID getDeliverToCompID()
      { QuickFix.DeliverToCompID value = new QuickFix.DeliverToCompID();
        getField(value); return value; }
      public bool isSet(QuickFix.DeliverToCompID field)
      { return isSetField(field); }
      public bool isSetDeliverToCompID()
      { return isSetField(128); }

      public void set(QuickFix.SecureDataLen value)
      { setField(value); }
      public QuickFix.SecureDataLen get(QuickFix.SecureDataLen  value)
      { getField(value); return value; }
      public QuickFix.SecureDataLen getSecureDataLen()
      { QuickFix.SecureDataLen value = new QuickFix.SecureDataLen();
        getField(value); return value; }
      public bool isSet(QuickFix.SecureDataLen field)
      { return isSetField(field); }
      public bool isSetSecureDataLen()
      { return isSetField(90); }

      public void set(QuickFix.SecureData value)
      { setField(value); }
      public QuickFix.SecureData get(QuickFix.SecureData  value)
      { getField(value); return value; }
      public QuickFix.SecureData getSecureData()
      { QuickFix.SecureData value = new QuickFix.SecureData();
        getField(value); return value; }
      public bool isSet(QuickFix.SecureData field)
      { return isSetField(field); }
      public bool isSetSecureData()
      { return isSetField(91); }

      public void set(QuickFix.MsgSeqNum value)
      { setField(value); }
      public QuickFix.MsgSeqNum get(QuickFix.MsgSeqNum  value)
      { getField(value); return value; }
      public QuickFix.MsgSeqNum getMsgSeqNum()
      { QuickFix.MsgSeqNum value = new QuickFix.MsgSeqNum();
        getField(value); return value; }
      public bool isSet(QuickFix.MsgSeqNum field)
      { return isSetField(field); }
      public bool isSetMsgSeqNum()
      { return isSetField(34); }

      public void set(QuickFix.SenderSubID value)
      { setField(value); }
      public QuickFix.SenderSubID get(QuickFix.SenderSubID  value)
      { getField(value); return value; }
      public QuickFix.SenderSubID getSenderSubID()
      { QuickFix.SenderSubID value = new QuickFix.SenderSubID();
        getField(value); return value; }
      public bool isSet(QuickFix.SenderSubID field)
      { return isSetField(field); }
      public bool isSetSenderSubID()
      { return isSetField(50); }

      public void set(QuickFix.SenderLocationID value)
      { setField(value); }
      public QuickFix.SenderLocationID get(QuickFix.SenderLocationID  value)
      { getField(value); return value; }
      public QuickFix.SenderLocationID getSenderLocationID()
      { QuickFix.SenderLocationID value = new QuickFix.SenderLocationID();
        getField(value); return value; }
      public bool isSet(QuickFix.SenderLocationID field)
      { return isSetField(field); }
      public bool isSetSenderLocationID()
      { return isSetField(142); }

      public void set(QuickFix.TargetSubID value)
      { setField(value); }
      public QuickFix.TargetSubID get(QuickFix.TargetSubID  value)
      { getField(value); return value; }
      public QuickFix.TargetSubID getTargetSubID()
      { QuickFix.TargetSubID value = new QuickFix.TargetSubID();
        getField(value); return value; }
      public bool isSet(QuickFix.TargetSubID field)
      { return isSetField(field); }
      public bool isSetTargetSubID()
      { return isSetField(57); }

      public void set(QuickFix.TargetLocationID value)
      { setField(value); }
      public QuickFix.TargetLocationID get(QuickFix.TargetLocationID  value)
      { getField(value); return value; }
      public QuickFix.TargetLocationID getTargetLocationID()
      { QuickFix.TargetLocationID value = new QuickFix.TargetLocationID();
        getField(value); return value; }
      public bool isSet(QuickFix.TargetLocationID field)
      { return isSetField(field); }
      public bool isSetTargetLocationID()
      { return isSetField(143); }

      public void set(QuickFix.OnBehalfOfSubID value)
      { setField(value); }
      public QuickFix.OnBehalfOfSubID get(QuickFix.OnBehalfOfSubID  value)
      { getField(value); return value; }
      public QuickFix.OnBehalfOfSubID getOnBehalfOfSubID()
      { QuickFix.OnBehalfOfSubID value = new QuickFix.OnBehalfOfSubID();
        getField(value); return value; }
      public bool isSet(QuickFix.OnBehalfOfSubID field)
      { return isSetField(field); }
      public bool isSetOnBehalfOfSubID()
      { return isSetField(116); }

      public void set(QuickFix.OnBehalfOfLocationID value)
      { setField(value); }
      public QuickFix.OnBehalfOfLocationID get(QuickFix.OnBehalfOfLocationID  value)
      { getField(value); return value; }
      public QuickFix.OnBehalfOfLocationID getOnBehalfOfLocationID()
      { QuickFix.OnBehalfOfLocationID value = new QuickFix.OnBehalfOfLocationID();
        getField(value); return value; }
      public bool isSet(QuickFix.OnBehalfOfLocationID field)
      { return isSetField(field); }
      public bool isSetOnBehalfOfLocationID()
      { return isSetField(144); }

      public void set(QuickFix.DeliverToSubID value)
      { setField(value); }
      public QuickFix.DeliverToSubID get(QuickFix.DeliverToSubID  value)
      { getField(value); return value; }
      public QuickFix.DeliverToSubID getDeliverToSubID()
      { QuickFix.DeliverToSubID value = new QuickFix.DeliverToSubID();
        getField(value); return value; }
      public bool isSet(QuickFix.DeliverToSubID field)
      { return isSetField(field); }
      public bool isSetDeliverToSubID()
      { return isSetField(129); }

      public void set(QuickFix.DeliverToLocationID value)
      { setField(value); }
      public QuickFix.DeliverToLocationID get(QuickFix.DeliverToLocationID  value)
      { getField(value); return value; }
      public QuickFix.DeliverToLocationID getDeliverToLocationID()
      { QuickFix.DeliverToLocationID value = new QuickFix.DeliverToLocationID();
        getField(value); return value; }
      public bool isSet(QuickFix.DeliverToLocationID field)
      { return isSetField(field); }
      public bool isSetDeliverToLocationID()
      { return isSetField(145); }

      public void set(QuickFix.PossDupFlag value)
      { setField(value); }
      public QuickFix.PossDupFlag get(QuickFix.PossDupFlag  value)
      { getField(value); return value; }
      public QuickFix.PossDupFlag getPossDupFlag()
      { QuickFix.PossDupFlag value = new QuickFix.PossDupFlag();
        getField(value); return value; }
      public bool isSet(QuickFix.PossDupFlag field)
      { return isSetField(field); }
      public bool isSetPossDupFlag()
      { return isSetField(43); }

      public void set(QuickFix.PossResend value)
      { setField(value); }
      public QuickFix.PossResend get(QuickFix.PossResend  value)
      { getField(value); return value; }
      public QuickFix.PossResend getPossResend()
      { QuickFix.PossResend value = new QuickFix.PossResend();
        getField(value); return value; }
      public bool isSet(QuickFix.PossResend field)
      { return isSetField(field); }
      public bool isSetPossResend()
      { return isSetField(97); }

      public void set(QuickFix.SendingTime value)
      { setField(value); }
      public QuickFix.SendingTime get(QuickFix.SendingTime  value)
      { getField(value); return value; }
      public QuickFix.SendingTime getSendingTime()
      { QuickFix.SendingTime value = new QuickFix.SendingTime();
        getField(value); return value; }
      public bool isSet(QuickFix.SendingTime field)
      { return isSetField(field); }
      public bool isSetSendingTime()
      { return isSetField(52); }

      public void set(QuickFix.OrigSendingTime value)
      { setField(value); }
      public QuickFix.OrigSendingTime get(QuickFix.OrigSendingTime  value)
      { getField(value); return value; }
      public QuickFix.OrigSendingTime getOrigSendingTime()
      { QuickFix.OrigSendingTime value = new QuickFix.OrigSendingTime();
        getField(value); return value; }
      public bool isSet(QuickFix.OrigSendingTime field)
      { return isSetField(field); }
      public bool isSetOrigSendingTime()
      { return isSetField(122); }

      public void set(QuickFix.XmlDataLen value)
      { setField(value); }
      public QuickFix.XmlDataLen get(QuickFix.XmlDataLen  value)
      { getField(value); return value; }
      public QuickFix.XmlDataLen getXmlDataLen()
      { QuickFix.XmlDataLen value = new QuickFix.XmlDataLen();
        getField(value); return value; }
      public bool isSet(QuickFix.XmlDataLen field)
      { return isSetField(field); }
      public bool isSetXmlDataLen()
      { return isSetField(212); }

      public void set(QuickFix.XmlData value)
      { setField(value); }
      public QuickFix.XmlData get(QuickFix.XmlData  value)
      { getField(value); return value; }
      public QuickFix.XmlData getXmlData()
      { QuickFix.XmlData value = new QuickFix.XmlData();
        getField(value); return value; }
      public bool isSet(QuickFix.XmlData field)
      { return isSetField(field); }
      public bool isSetXmlData()
      { return isSetField(213); }

      public void set(QuickFix.MessageEncoding value)
      { setField(value); }
      public QuickFix.MessageEncoding get(QuickFix.MessageEncoding  value)
      { getField(value); return value; }
      public QuickFix.MessageEncoding getMessageEncoding()
      { QuickFix.MessageEncoding value = new QuickFix.MessageEncoding();
        getField(value); return value; }
      public bool isSet(QuickFix.MessageEncoding field)
      { return isSetField(field); }
      public bool isSetMessageEncoding()
      { return isSetField(347); }

      public void set(QuickFix.LastMsgSeqNumProcessed value)
      { setField(value); }
      public QuickFix.LastMsgSeqNumProcessed get(QuickFix.LastMsgSeqNumProcessed  value)
      { getField(value); return value; }
      public QuickFix.LastMsgSeqNumProcessed getLastMsgSeqNumProcessed()
      { QuickFix.LastMsgSeqNumProcessed value = new QuickFix.LastMsgSeqNumProcessed();
        getField(value); return value; }
      public bool isSet(QuickFix.LastMsgSeqNumProcessed field)
      { return isSetField(field); }
      public bool isSetLastMsgSeqNumProcessed()
      { return isSetField(369); }

      public void set(QuickFix.NoHops value)
      { setField(value); }
      public QuickFix.NoHops get(QuickFix.NoHops  value)
      { getField(value); return value; }
      public QuickFix.NoHops getNoHops()
      { QuickFix.NoHops value = new QuickFix.NoHops();
        getField(value); return value; }
      public bool isSet(QuickFix.NoHops field)
      { return isSetField(field); }
      public bool isSetNoHops()
      { return isSetField(627); }

      public class NoHops: QuickFix.Group
      {
      public NoHops() : base(627,628,message_order ) {}
      static int[] message_order = new int[] {628,629,630,0};
        public void set(QuickFix.HopCompID value)
        { setField(value); }
        public QuickFix.HopCompID get(QuickFix.HopCompID  value)
        { getField(value); return value; }
        public QuickFix.HopCompID getHopCompID()
        { QuickFix.HopCompID value = new QuickFix.HopCompID();
          getField(value); return value; }
        public bool isSet(QuickFix.HopCompID field)
        { return isSetField(field); }
        public bool isSetHopCompID()
        { return isSetField(628); }

        public void set(QuickFix.HopSendingTime value)
        { setField(value); }
        public QuickFix.HopSendingTime get(QuickFix.HopSendingTime  value)
        { getField(value); return value; }
        public QuickFix.HopSendingTime getHopSendingTime()
        { QuickFix.HopSendingTime value = new QuickFix.HopSendingTime();
          getField(value); return value; }
        public bool isSet(QuickFix.HopSendingTime field)
        { return isSetField(field); }
        public bool isSetHopSendingTime()
        { return isSetField(629); }

        public void set(QuickFix.HopRefID value)
        { setField(value); }
        public QuickFix.HopRefID get(QuickFix.HopRefID  value)
        { getField(value); return value; }
        public QuickFix.HopRefID getHopRefID()
        { QuickFix.HopRefID value = new QuickFix.HopRefID();
          getField(value); return value; }
        public bool isSet(QuickFix.HopRefID field)
        { return isSetField(field); }
        public bool isSetHopRefID()
        { return isSetField(630); }

      };
  };

  public class Trailer : QuickFix.Message.Trailer
  {
    Trailer(QuickFix.Message message) : base(message) {}

      public void set(QuickFix.SignatureLength value)
      { setField(value); }
      public QuickFix.SignatureLength get(QuickFix.SignatureLength  value)
      { getField(value); return value; }
      public QuickFix.SignatureLength getSignatureLength()
      { QuickFix.SignatureLength value = new QuickFix.SignatureLength();
        getField(value); return value; }
      public bool isSet(QuickFix.SignatureLength field)
      { return isSetField(field); }
      public bool isSetSignatureLength()
      { return isSetField(93); }

      public void set(QuickFix.Signature value)
      { setField(value); }
      public QuickFix.Signature get(QuickFix.Signature  value)
      { getField(value); return value; }
      public QuickFix.Signature getSignature()
      { QuickFix.Signature value = new QuickFix.Signature();
        getField(value); return value; }
      public bool isSet(QuickFix.Signature field)
      { return isSetField(field); }
      public bool isSetSignature()
      { return isSetField(89); }

      public void set(QuickFix.CheckSum value)
      { setField(value); }
      public QuickFix.CheckSum get(QuickFix.CheckSum  value)
      { getField(value); return value; }
      public QuickFix.CheckSum getCheckSum()
      { QuickFix.CheckSum value = new QuickFix.CheckSum();
        getField(value); return value; }
      public bool isSet(QuickFix.CheckSum field)
      { return isSetField(field); }
      public bool isSetCheckSum()
      { return isSetField(10); }

  };

  public class Message : QuickFix.Message
  {
    public Message() : base(new QuickFix.BeginString("FIX.4.4"))
    {
      m_header = new Header(this);
      m_trailer = new Trailer(this);
    }

    public Message( QuickFix.MsgType msgType ) : base(new QuickFix.BeginString("FIX.4.4"), msgType)
    {
      m_header = new Header(this);
      m_trailer = new Trailer(this);
    }

    public new Header getHeader() { return (Header)(base.getHeader()); }
    public new Trailer getTrailer() { return (Trailer)(base.getTrailer()); }
  };

}

